test/battle/ai.c

```diff
@@ -699,6 +699,19 @@ AI_SINGLE_BATTLE_TEST("AI_FLAG_SMART_MON_CHOICES: Post-KO switches prioritize of
     }
 }
 
+AI_SINGLE_BATTLE_TEST("AI_FLAG_SMART_MON_CHOICES: U-Turn will send out Ace Mon if it's the only one remaining")
```

Comment:
Does this test prove the fix? It is just one move with one mon in the back

Reply:
Yeah actually, with the bug present it tries to switch in index -1 lol